geomstats/information_geometry/geometric.py

```diff
@@ -0,0 +1,393 @@
+            (1 - atol) * gs.cast(point > 1 - atol, point.dtype)
+            + atol * gs.cast(point < atol, point.dtype),
+            point,
+        )
+
+    def sample(self, point, n_samples=1):
+        """Sample from the geometric distribution.
+
+        Sample from the geometric distribution with parameter provided
+        by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point representing an geometric distribution.
+        n_samples : int
+            Number of points to sample with each parameter in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples]
+            Sample from geometric distributions.
+        """
+
+        def _sample(param):
+            return geom.rvs(param, size=n_samples)
+
+        if point.ndim > 1:
+            return gs.array([_sample(point_) for point_ in point])
+
+        return gs.array(_sample(point))
+
+    def point_to_pdf(self, point):
+        """Compute pmf associated to point.
+
+        Compute the probability mass function of the geometric
+        distribution with parameters provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point representing an geometric distribution (scale).
+
+        Returns
+        -------
+        pdf : function
+            Probability mass function of the geometric distribution with
+            scale parameter provided by point.
+        """
+
+        def pmf(k):
+            """Generate parameterized function for geometric pmf.
+
+            Compute the probability mass function of the geometric
+            distribution with parameters provided by point.
+
+            Parameters
+            ----------
+            k : array-like, shape=[n_points,]
+                Point representing an geometric distribution (lambda).
+
+            Returns
+            -------
+            pmf_at_k : array-like, shape=[..., n_points]
+                Probability mass function of the geometric distribution with
+                parameters provided by point.
+            """
+            k = gs.reshape(gs.array(k), (-1,))
+            return (1 - point) ** (k - 1) * point
+
+        return pmf
+
+
+class GeometricMetric(RiemannianMetric):
+    """Class for the Fisher information metric on geometric distributions.
+
+    References
+    ----------
+    .. [AM1981] Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
+        Sankhyā: The Indian Journal of Statistics, Series A, 345-365.
+    """
+
+    def __init__(self):
+        super().__init__(dim=1)
+
+    def squared_dist(self, point_a, point_b, **kwargs):
+        """Compute squared distance associated with the geometric metric.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., 1]
+            Point representing an geometric distribution (lambda parameter).
```

Comment:
lambda is non standard for the parameter of a geometric distribution, I'm not sure it helps.